server/shadow/shadow_server.c

```diff
@@ -510,16 +536,50 @@ int shadow_server_start(rdpShadowServer* server)
 		return -1;
 	}
 
-	if (!server->ipcSocket)
-		status = server->listener->Open(server->listener, NULL, (UINT16)server->port);
-	else
-		status = server->listener->OpenLocal(server->listener, server->ipcSocket);
+	/* Bind to TCP port if nothing is specified, otherwise bind to everything specified */
```

Comment:
With your refactoring, this comment is no longer valid, since it will only bind a TCP socket _or_ a local socket